misk/src/test/kotlin/misk/client/ClientMetricsInterceptorTest.kt

```diff
@@ -1,123 +1,117 @@
+    mockWebServer.enqueue(MockResponse().setResponseCode(202).setBody("{}"))
+    mockWebServer.enqueue(MockResponse().setResponseCode(403).setBody("{}"))
+    mockWebServer.enqueue(MockResponse().setResponseCode(404).setBody("{}"))
+    mockWebServer.enqueue(MockResponse().setResponseCode(503).setBody("{}"))
     assertThat(client.ping(AppRequest(200)).execute().code()).isEqualTo(200)
     assertThat(client.ping(AppRequest(200)).execute().code()).isEqualTo(200)
     assertThat(client.ping(AppRequest(202)).execute().code()).isEqualTo(202)
     assertThat(client.ping(AppRequest(403)).execute().code()).isEqualTo(403)
     assertThat(client.ping(AppRequest(404)).execute().code()).isEqualTo(404)
     assertThat(client.ping(AppRequest(503)).execute().code()).isEqualTo(503)
-  }
 
-  @Test
-  fun responseCodes() {
-    requestDuration.record(1.0, "pinger.ping", "all")
-    assertThat(requestDuration.count("pinger.ping", "all")).isEqualTo(7)
-    requestDuration.record(1.0, "pinger.ping", "202")
-    assertThat(requestDuration.count("pinger.ping", "202")).isEqualTo(2)
-    requestDuration.record(1.0, "pinger.ping", "4xx")
-    assertThat(requestDuration.count("pinger.ping", "4xx")).isEqualTo(3)
-    requestDuration.record(1.0, "pinger.ping", "404")
-    assertThat(requestDuration.count("pinger.ping", "404")).isEqualTo(2)
-    requestDuration.record(1.0, "pinger.ping", "403")
-    assertThat(requestDuration.count("pinger.ping", "403")).isEqualTo(2)
-    requestDuration.record(1.0, "pinger.ping", "5xx")
-    assertThat(requestDuration.count("pinger.ping", "403")).isEqualTo(2)
-    requestDuration.record(1.0, "pinger.ping", "503")
-    assertThat(requestDuration.count("pinger.ping", "503")).isEqualTo(2)
-  }
 
-  class TestModule : KAbstractModule() {
-    override fun configure() {
-      install(PrometheusHistogramRegistryModule())
-      install(WebTestingModule())
-      install(WebActionModule.create<PingAction>())
+    SoftAssertions.assertSoftly { softly ->
+      softly.assertThat(requestDuration.count("pinger.ping", "all")).isEqualTo(6)
+      softly.assertThat(requestDuration.count("pinger.ping", "202")).isEqualTo(1)
+      softly.assertThat(requestDuration.count("pinger.ping", "4xx")).isEqualTo(2)
+      softly.assertThat(requestDuration.count("pinger.ping", "404")).isEqualTo(1)
+      softly.assertThat(requestDuration.count("pinger.ping", "403")).isEqualTo(1)
+      softly.assertThat(requestDuration.count("pinger.ping", "403")).isEqualTo(1)
+      softly.assertThat(requestDuration.count("pinger.ping", "503")).isEqualTo(1)
     }
   }
 
-  interface Pinger {
-    @POST("/ping")
-    @Headers(
-            "Accept: " + MediaTypes.APPLICATION_JSON,
-            "Content-type: " + MediaTypes.APPLICATION_JSON)
-    fun ping(@Body request: AppRequest): Call<AppResponse>
-  }
+  @Test
+  fun timeouts() {
+    assertThatExceptionOfType(ExecutionException::class.java).isThrownBy {
+      client.ping(AppRequest(200)).execute().code()
```

Comment:
interesting, where do we setup the mock server to timeout?

Reply:
You don't. The mock server only returns the enqueued requests. If you don't queue anything, it won't return, causing a timeout.